telethon/telegram_client.py

```diff
@@ -239,7 +246,7 @@ def __init__(self, session, api_id, api_hash,
                 offset_peer=offset_peer,
                 limit=need if need < float('inf') else 0
             ))
-            if not r.dialogs:
+            if not r or not r.dialogs:
```

Comment:
So this could break even without having read a single dialog when there were. Not good!